src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -267,12 +274,25 @@ private void SolutionManager_CacheUpdated(object sender, NuGetEventArgs<string>
                     // We also refresh the UI if projectFullPath is not present.
                     if (!projectContainsFullPath || projectFullName == eventProjectFullName)
                     {
-                        Refresh();
+                        RefreshIfNeeded();
                     }
                 }
             }
         }
 
+        private void RefreshIfNeeded()
```

Comment:
It's needed whether or not an action is execution.  The difference is when.

How about `RefreshWhenNotExecutingAction`?